certtools/keys.go

```diff
@@ -76,6 +99,40 @@ func GetKey(eng *url.URL, file, newkey string) (crypto.PrivateKey, error) {
 	return nil, fmt.Errorf("no key mechanism specified")
 }
 
+func verifyExistingKey(file string, existingKey crypto.PrivateKey, newkey string) (crypto.PrivateKey, error) {
+	//desired type specified, verify it
+	specs := strings.Split(newkey, ":")
+	switch t := existingKey.(type) {
+	case *ecdsa.PrivateKey:
+		if specs[0] != "ec" {
+			return nil, fmt.Errorf("detected existing key [%s] but was of the wrong type, expected an EC key", file)
+		}
+
+		if t.Curve != CURVES[specs[1]] {
+			return nil, fmt.Errorf("detected existing key [%s] but was of the wrong curve type: %s, expected: %s", file, t.Curve.Params().Name, specs[1])
+		}
+
+		return existingKey, nil
+
+	case *rsa.PrivateKey:
+		if specs[0] != "rsa" {
+			return nil, fmt.Errorf("detected existing key [%s] but was of the wrong type, expected an RSA key", file)
+		}
+		bitSize, err := strconv.Atoi(specs[1])
```

Comment:
if the specs[1] isn't a number here, that's "fine"?

Reply:
error is handled below.